watsononlinestore/watson_online_store.py

```diff
@@ -292,6 +294,23 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        num_matches = 0
+        filtered_results = []
+        if self.discovery_score_filter:
+            results = (results for results in discovery_response['results']
+                      if 'results' in discovery_response)
```

Comment:
The original code was better. To bring in list comprehension this one has some problems.
1. It should be results = (result for result in...
2. That if is called on each result. So we'll get the KeyError that it was intended to avoid
I'm not sure if building a new list with comprehension helps here.